server/parameters/util/HelpEntry.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.vaticle.typedb.core.server.parameters.util;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+import static com.vaticle.typedb.core.common.iterator.Iterators.iterate;
+
+public abstract class HelpEntry {
+
+    final String scopedName;
+    final String description;
+
+    private HelpEntry(String scopedName, String description) {
+        this.scopedName = scopedName;
+        this.description = description;
+    }
+
+    public static abstract class Yaml extends HelpEntry {
+
+        private Yaml(String scopedName, String description) {
+            super(scopedName, description);
+        }
+
+        public static class Grouped extends Yaml {
```

Comment:
ok on reading down in the usages more it feels strange to have a single subtype of `Yaml` for `Grouped`. Either we should create a second "simple" or "leaf"  HelpEntry for Yaml, that is used in the `YamlParser`, or flatten the `Grouped` into the `Yaml` with a decent name. 

For example, the content of `Yaml.Grouped` actually itself has nothing yaml-specific, but only that is holding a group of entries.

Reply:
> For example, the content of Yaml.Grouped actually itself has nothing yaml-specific, but only that is holding a group of entries.

Actually, I deliberately put them under Yaml because it is Yaml specific - take a look at how we render the String with `.` separating the children.